suite-native/app/e2e/tests/deeplinkPopup.test.ts

```diff
@@ -0,0 +1,162 @@
+import { onCoinEnablingInit } from '../pageObjects/coinEnablingActions';
+import { onHome } from '../pageObjects/homeActions';
+
+const SERVER_PORT = 8080;
+const SERVER_URL = `http://localhost:${SERVER_PORT}`;
+
+let server: http.Server | undefined;
+
+const openUriScheme = (url: string, platformToOpen: 'android') => {
+    const command = `npx uri-scheme open '${url.replace(/'/g, '')}' --${platformToOpen}`;
+
+    exec(command, (err, stdout, stderr) => {
+        if (err) {
+            console.error(err);
+
+            return;
+        }
+        // eslint-disable-next-line no-console
+        console.info(stdout);
+        console.error(stderr);
+    });
+};
+
+const reverseServerPort = (port: number) => {
+    return new Promise((resolve, reject) => {
+        exec(
+            `/usr/local/bin/platform-tools/adb reverse tcp:${port} tcp:${port}`,
+            (err, stdout, stderr) => {
+                if (err) {
+                    console.error(err);
+
+                    return reject(err);
+                }
+                // eslint-disable-next-line no-console
+                console.info(stdout);
+                console.error(stderr);
+                resolve(stdout);
+            },
+        );
+    });
+};
+
+describe('Deeplink connect popup.', () => {
+    beforeAll(async () => {
+        server = http.createServer((req, res) => {
+            if (req.url) {
+                const url = new URL(req.url, SERVER_URL);
+                TrezorConnect.handleDeeplink(url.href);
+                res.statusCode = 200;
+                res.setHeader('Content-Type', 'text/plain');
+                res.end('Callback URL received successfully!\n');
+            }
+        });
+
+        server.listen(SERVER_PORT, 'localhost', () => {
+            // eslint-disable-next-line no-console
+            console.info(`Server running at ${SERVER_URL}`);
+        });
+
+        await prepareTrezorEmulator();
+        await openApp({ newInstance: true });
+        await onOnboarding.finishOnboarding();
+
+        await onCoinEnablingInit.waitForScreen();
+        await onCoinEnablingInit.enableNetwork('regtest');
+        await onCoinEnablingInit.clickOnConfirmButton();
+
+        await waitFor(element(by.id('skip-view-only-mode')))
+            .toBeVisible()
+            .withTimeout(10000); // communication between connected Trezor and app takes some time.
+
+        await element(by.id('skip-view-only-mode')).tap();
+
+        // This `TrezorConnect` instance here is pretending to be the integrator or @trezor/connect-mobile
+        await TrezorConnect.init({
+            manifest: {
+                email: 'developer@xyz.com',
+                appUrl: 'http://your.application.com',
+            },
+            deeplinkOpen: url => {
+                openUriScheme(url, 'android');
+            },
+            deeplinkCallbackUrl: `${SERVER_URL}/connect/`,
+            connectSrc: 'https://dev.suite.sldev.cz/connect/9.1.2/',
```

Comment:
btw this link is alive?

Reply:
No, it actually works because it satisfy the intentFilters https://github.com/trezor/trezor-suite/blob/develop/suite-native/app/app.config.ts#L200

Not critical for tests, but yes, better to have real link, changed:

https://github.com/trezor/trezor-suite/pull/14667/commits/dd885269503998d6a820532bddf455f69ff34c54

So it works also with the nice fallback in web popup.